ambry-router/src/main/java/com.github.ambry.router/GetBlobInfoOperation.java

```diff
@@ -370,5 +433,35 @@ private void checkAndMaybeComplete() {
       NonBlockingRouter.completeOperation(null, getOperationCallback, operationResult, e);
     }
   }
+
+  /**
+   * @return true if the content is being decrypted
+   */
+  private boolean onDecryptMode() {
+    return state == State.Decrypting;
+  }
+
+  /**
+   * @return true if the operation is complete.
+   */
+  private boolean isComplete() {
+    return state == State.Complete;
+  }
+
+  /**
+   * Different states of the operation.
+   */
+  enum State {
+    /**
+     * Any GetBlobInfoOperation starts with this mode
+     */
+    Ready, /**
+     * If blob is encrypted, the operation moves to this mode and stays until the decrypt job callback results are processed
+     */
+    Decrypting, /**
```

Comment:
Seems like previously we did not have any states, can't that be maintained if we handled `decrypting` as an internal detail?

Reply:
commented above. I feel that these depict the state of the chunk and also instead of having 2 to 3 diff variables, its better to use one variable to check the state. May be router or poll() may not be interested in some of the states. But my opinion is that, its okay to have some extra states if its been used only internally since "state" as such is used to track a GetChunk's state.